Armor Cleanup + Changes

TC-DarkSlayer opened the pull request:
- Cleans up FleshHit.as
- Adds new armor (Buckethead/Barrels) very weak though.
- Keg wearers will explode through explosives and fire
- Kegs can now be lit by lighters
- NOTE: this uses pr: Equipment Cleanup

DARK-DarkSlayer committed:
Equipment Cleanup

- Cleans up Equipment.as, reduced total lines by 200+

DARK-DarkSlayer committed:
Equipment Cleanup - NameChanges

- Required for main Equipment Cleanup
- Changes getSprite()'s calling name.
- added .png for navigation to parachutepack_effect.as

DARK-DarkSlayer committed:
Added Command: !armor

- Command gives military armor for head, torso, boots.
- This is a faster way for devs to spawn in armor

DARK-DarkSlayer committed:
Armor Cleanup

- Variables are now easier so that you don't have to change every single one when making new armor.
- Makes armor creating easier.

DARK-DarkSlayer committed:
Bucket/Barrel Armor

- Buckets/Barrels have low durability, don't block bullets, and low protection for the first few hits
- People wearing kegs after being exposed to fire/bombs will explode themselves

DARK-DarkSlayer committed:
Lighters lite Kegs

- Lighters now lite kegs properly

DARK-DarkSlayer committed:
Keg armor HP change

- Makes keg armor hp a little bit more than bucket
5 to 7 hp

Vam-Jam commented on TFlippy_TerritoryControl_Core_Dev/Entities/Equipment/Equipment.as:
```diff
@@ -66,156 +66,64 @@ void onCreateInventoryMenu(CBlob@ this, CBlob@ forBlob, CGridMenu@ gridmenu)
 		CBlob@ caller = getBlobByNetworkID(callerID);
-		if (caller is null)
-			return;
+		if (caller is null) return;
```
if (caller is null) return;
space
next if statement

Vam-Jam commented on TFlippy_TerritoryControl_Core_Dev/Entities/Equipment/Equipment.as:
```diff
@@ -66,156 +66,64 @@ void onCreateInventoryMenu(CBlob@ this, CBlob@ forBlob, CGridMenu@ gridmenu)
-		if (caller is null)
-			return;
+		if (caller is null) return;
+		if (caller.get_string("equipment_torso") != "" && cmd == this.getCommandID("equip_torso")) removeTorso(caller, caller.get_string("equipment_torso"));
```
far too long

Vam-Jam commented on TFlippy_TerritoryControl_Core_Dev/Entities/Equipment/Equipment.as:
```diff
@@ -233,98 +141,46 @@ void addHead(CBlob@ playerblob, string headname)	//Here you need to add head ove
 	playerblob.Tag(headname);
 	playerblob.set_string("reload_script", headname);
-	if(headname != "pumpkin" && headname != "bucket") playerblob.AddScript(headname+"_effect.as");
+	if(headname != "pumpkin") playerblob.AddScript(headname+"_effect.as");
```
space
if space (headname
space

Vam-Jam commented on TFlippy_TerritoryControl_Core_Dev/Entities/Equipment/Equipment.as:
```diff
@@ -66,156 +66,64 @@ void onCreateInventoryMenu(CBlob@ this, CBlob@ forBlob, CGridMenu@ gridmenu)
-			}
-		}
-		else
+		if(item !is null)
```
if space (item !is null)

Vam-Jam commented on TFlippy_TerritoryControl_Core_Dev/Entities/Equipment/Equipment.as:
```diff
@@ -233,98 +141,46 @@ void addHead(CBlob@ playerblob, string headname)	//Here you need to add head ove
-			playerblob.set_u8("last head", playerblob.get_u8("override head"));
-		else	
-			playerblob.set_u8("last head", playerblob.getHeadNum());
+		if(playerblob.get_u8("override head") != 0) playerblob.set_u8("last head", playerblob.get_u8("override head"));
```
if space (

Vam-Jam commented on TFlippy_TerritoryControl_Core_Dev/Entities/Equipment/Equipment.as:
```diff
@@ -233,98 +141,46 @@ void addHead(CBlob@ playerblob, string headname)	//Here you need to add head ove
-
-	if(headname == "pumpkin")
-		playerblob.set_u8("override head", 108);
+	if(headname == "scubagear") playerblob.set_u8("override head", 88);
```
if space (

Vam-Jam commented on TFlippy_TerritoryControl_Core_Dev/Entities/Equipment/Equipment.as:
```diff
@@ -233,98 +141,46 @@ void addHead(CBlob@ playerblob, string headname)	//Here you need to add head ove
 	{
-		if(headname == "militaryhelmet")			//need to be after creating blob, bcos it sets hp to it
+		CBlob@ oldeq = server_CreateBlob(headname, playerblob.getTeamNum(), playerblob.getPosition());
+		if(headname == "militaryhelmet")	//need to be after creating blob, bcos it sets hp to it
```
if space (

Vam-Jam commented on TFlippy_TerritoryControl_Core_Dev/Entities/Equipment/Equipment.as:
```diff
@@ -233,98 +141,46 @@ void addHead(CBlob@ playerblob, string headname)	//Here you need to add head ove
-			playerblob.server_PutInInventory(oldeq);
 		}
-		else
+		else if(headname == "bucket")
```
if space (

Vam-Jam commented on TFlippy_TerritoryControl_Core_Dev/Entities/Equipment/Equipment.as:
```diff
@@ -469,16 +272,16 @@ void onDie(CBlob@ this)
 		if (this.get_string("equipment_boots") != "")
 		{
-			server_CreateBlob(this.get_string("equipment_boots"), this.getTeamNum(), this.getPosition());
+			if(this.get_string("equipment_boots") == "combatboots")
```
if space

Vam-Jam commented on TFlippy_TerritoryControl_Core_Dev/Entities/Equipment/Equipment.as:
```diff
@@ -469,16 +272,16 @@ void onDie(CBlob@ this)
 		{
-			server_CreateBlob(this.get_string("equipment_boots"), this.getTeamNum(), this.getPosition());
+			if(this.get_string("equipment_boots") == "combatboots")
+			{ }
```
is this suppose to be empty?

TC-DarkSlayer replied:
I'd assume goldenguy wanted to do
if (this.get_string("equipment_boots") != "combatboots")

Vam-Jam commented on TFlippy_TerritoryControl_Core_Dev/Entities/FleshHit.as:
```diff
@@ -121,89 +87,113 @@ f32 onHit(CBlob@ this, Vec2f worldPoint, Vec2f velocity, f32 damage, CBlob@ hitt
-				case HittersTC::shotgun:
-					ratio = ratio * 0.85f;
+
+				case HittersTC::shotgun: ratio *= 0.85f;
```
case shotgun:
 ration
 break;
 
imo keep this consistent because you already have 1 like it

Vam-Jam commented on TFlippy_TerritoryControl_Core_Dev/Entities/FleshHit.as:
```diff
@@ -121,89 +87,113 @@ f32 onHit(CBlob@ this, Vec2f worldPoint, Vec2f velocity, f32 damage, CBlob@ hitt
+			{
+				this.set_f32("keg_explode", getGameTime() + (30.0f * 1.0f));
+				this.SetLightRadius(this.get_f32("explosive_radius") * 0.5f);
+				this.getSprite().SetEmitSound("/Sparkle.ogg");
```
instead of this.getSprite() 5 times, use CSprite@ sprite = this.getSprite()

Vam-Jam commented on TFlippy_TerritoryControl_Core_Dev/Entities/FleshHit.as:
```diff
@@ -121,89 +87,113 @@ f32 onHit(CBlob@ this, Vec2f worldPoint, Vec2f velocity, f32 damage, CBlob@ hitt
-				case HittersTC::shotgun:
-					ratio = ratio * 0.85f;
+
+				case HittersTC::shotgun: ratio *= 0.85f;
```
same thing here

Vam-Jam commented on TFlippy_TerritoryControl_Core_Dev/Entities/FleshHit.as:
```diff
@@ -121,89 +87,113 @@ f32 onHit(CBlob@ this, Vec2f worldPoint, Vec2f velocity, f32 damage, CBlob@ hitt
+					ratio *= 0.10f;
+					break;
+
+				default: ratio *= 0.20f;
```
and here

Vam-Jam commented on TFlippy_TerritoryControl_Misc_Dev/Entities/Equipment/Barrel/Keg.as:
```diff
@@ -0,0 +1,128 @@
+
+	sprite.SetZ(-10);
+
+	CSpriteLayer@ fuse = this.getSprite().addSpriteLayer("fuse", "Keg.png" , 16, 16, 0, 0);
```
replace this.getSprite() with sprite

Vam-Jam commented on TFlippy_TerritoryControl_Misc_Dev/Entities/Equipment/Barrel/keg_effect.as:
```diff
@@ -26,4 +35,75 @@ void onTick(CBlob@ this)
+void onDie(CBlob@ this)
+{
+	DoExplosion(this);
+	if (isServer() && (this.get_f32("keg_explode") == 0.0f))
```
space 
if
{ 
}
space

Vam-Jam commented on TFlippy_TerritoryControl_Misc_Dev/Entities/Equipment/BucketHead/bucket_effect.as:
```diff
@@ -0,0 +1,39 @@
+        UpdateScript(this);
+}
+
+void UpdateScript(CBlob@ this)
```
empty

Vam-Jam commented on TFlippy_TerritoryControl_Misc_Dev/Entities/Equipment/BucketHead/bucket_effect.as:
```diff
@@ -0,0 +1,39 @@
+ 
+void onTick(CBlob@ this)
+{
+    if(this.get_string("reload_script") == "bucket")
```
if space (

Vam-Jam commented on TFlippy_TerritoryControl_Misc_Dev/Entities/Equipment/BucketHead/bucket_effect.as:
```diff
@@ -0,0 +1,39 @@
+        this.set_string("reload_script", "");
+    }
+   
+    if(this.get_f32("bucket_health") >= 5.0f)
```
space

Vam-Jam commented on TFlippy_TerritoryControl_Misc_Dev/Entities/Equipment/BucketHead/bucket_effect.as:
```diff
@@ -0,0 +1,39 @@
+    if (isServer())
+    {
+        CBlob@ item = server_CreateBlob("bucket", this.getTeamNum(), this.getPosition());
+        if(item !is null) item.set_f32("health", this.get_f32("bucket_health"));
```
space

DARK-DarkSlayer committed:
Cleaning Up

- adds a space to if( to if (
- removes some empty methods

DARK-DarkSlayer committed:
More Cleanup

More Cleanup

DARK-DarkSlayer committed:
Bug fixing the cleanup

- resolving removing scripts left unmatching signatures.

DARK-DarkSlayer committed:
Combat Boots Equipment

- For in the future, when someone wants to make boots be a viable armor
- creates the blob while retaining its health/durability on death

Vam-Jam commented on TFlippy_TerritoryControl_Misc_Dev/Entities/Equipment/Barrel/keg_effect.as:
```diff
@@ -1,29 +1,110 @@
 	}
+
+	if (this.get_f32("keg_health") >= 7.0f)
+    {
```
indent is incorrect

Vam-Jam commented on TFlippy_TerritoryControl_Misc_Dev/Entities/Equipment/Barrel/keg_effect.as:
```diff
@@ -1,29 +1,110 @@
+        this.getSprite().PlaySound("woodheavyhit1");
+        this.set_string("equipment_head", "");
+        this.set_f32("keg_health", 0.0f);
+		if (this.getSprite().getSpriteLayer("keg") !is null) this.getSprite().RemoveSpriteLayer("keg");
```
same here

DARK-DarkSlayer committed:
Indent Fix

- seems that I copied someone's spaces for indent when copy and pasting

Vam-Jam committed:
Merge branch 'master' into DarkieChanges
